### PR TITLE
Architecture diagrams of HttpServer and Connection lifecycle

### PR DESCRIPTION
**Pull request by dunnock**
Hi, I have spent some time to dig into actix-service and actix-web and decided to write it down while it is still fresh knowledge. This will add 2 pages on the left menu in docs:

HttpServer -  [content/docs/http_server_init.md](https://github.com/dunnock/actix-website/blob/arch/content/docs/conn_lifecycle.md)

Connection lifecycle - [content/docs/conn_lifecycle.md ](https://github.com/dunnock/actix-website/blob/arch/content/docs/conn_lifecycle.md)

With respective diagrams:
 - static/img/diagrams/*.mmd - diagrams in mermaid format
 - static/img/diagrams/*.svg - compiled diagrams

Diagrams cover some private structs too, which are required to understand lifecycle.

### CODE REVIEW

**Comment by robjtede** [comment]
This looks valuable. Can we get some READMEs of what tools are needed to generate the SVGs from .mmd files.

(and capitalize the L in Lifecycle of the page title)

File: content/docs/http_server_init.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: Http Server Initialization
+menu: docs_architecture
+weight: 10
+---
+
+## Architecture overview
+
+Below is a diagram of HttpServer initalization, which happens on the following code
+```
```
Comment (JohnTitor) [inline content/docs/http_server_init.md]:
```suggestion

```rust
```

File: content/docs/conn_lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+After Server has started with all connections listening, there are two main loops responsible for processing incoming client connections on established sockets Accept and Worker. Application level protocol processing happens in separate loop spawned from Worker after connection accepted.
+
+![](/img/diagrams/connection_overview.svg)
+
+## Accept loop in more detail
+
+![](/img/diagrams/connection_accept.svg)
+
+Most of code implementation resides in [actix-service](https://crates.io/crates/actix-service) crate.
```
Comment (JohnTitor) [inline content/docs/conn_lifecycle.md]: The link should be the reference-style link, like:

```
[`actix-service`][actixservice]
```

Should place links at the end of the file and add backticks around the word (also remove trailing spaces). Below are the same.

File: content/docs/conn_lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+title: Connection Lifecycle
+menu: docs_architecture
+weight: 20
+---
+
+
+# Architecture overview
+
+After Server has started with all connections listening, there are two main loops responsible for processing incoming client connections on established sockets Accept and Worker. Application level protocol processing happens in separate loop spawned from Worker after connection accepted.
```
Comment (JohnTitor) [inline content/docs/conn_lifecycle.md]: This line is too long, split it into several lines.

File: README.md [hunk 4]
```diff
@@ -21,6 +21,18 @@ hugo server
 
 Then visit [http://localhost:1313](http://localhost:1313).
 
+## Updating diagrams
+
+Diagrams are located under [/static/css/img/diagrams/](https://github.com/actix/actix-website/tree/master/static/img/diagrams) and built with [Mermaid CLI](https://github.com/mermaidjs/mermaid.cli).
+
+For instance to edit `connection_overview` diagram:
+```
```
[the remark this answers is not in the document]
Comment (dunnock) [reply README.md]: Fixed

File: content/docs/http_server_init.md (hunk 1 above)
Comment (dunnock) [reply content/docs/http_server_init.md]: Fixed

File: content/docs/conn_lifecycle.md (hunk 2 above)
Comment (dunnock) [reply content/docs/conn_lifecycle.md]: Fixed, also added links to Accept, Worker and Dispatcher. Page does not display links at the bottom though, same as on other pages.

File: content/docs/conn_lifecycle.md (hunk 3 above)
Comment (dunnock) [reply content/docs/conn_lifecycle.md]: Fixed, rephrased too

**Comment by JohnTitor** [comment]
Thanks!

**Pull request merged**